syft/pkg/cataloger/cataloger.go

```diff
@@ -114,10 +116,15 @@ func AllCatalogers(cfg Config) []Cataloger {
 		cpp.NewConanfileCataloger(),
 		portage.NewPortageCataloger(),
 		haskell.NewHackageCataloger(),
-	}, cfg.Catalogers)
+	}, cfg)
 }
 
+// RequestedAllCatalogers returns true if all Catalogers have been requested. Takes into account cfg.ExternalSourcesEnabled
 func RequestedAllCatalogers(cfg Config) bool {
+	if !cfg.ExternalSourcesEnabled {
+		return false
+	}
```

Comment:
this probably should be a function of running the loop on all catalogers that exist and checking if they `UsesExternalSources()`